src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -1515,7 +1541,14 @@ private boolean isSourceChanged( CompilerConfiguration compilerConfiguration, Co
         {
             for ( File f : staleSources )
             {
-                getLog().debug( "Stale source detected: " + f.getAbsolutePath() );
+                if ( showCompilationChanges )
```

Comment:
we cannot see it in this GH ui diff but we go in this path only `if ( getLog().isDebugEnabled() )` this must be changed to something such 
`if ( !staleSources.isEmpty() && (getLog().isDebugEnabled() ||  showCompilationChanges))`
I guess the idea is to display those files when using `-Dmaven.compiler.showCompilationChanges=true` but without using full debug with `-X`